lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2386,12 +2386,18 @@ public function getCustomTypeDeclarationSQL(array $columnDef)
      * Obtains DBMS specific SQL code portion needed to set an index
      * declaration to be used in statements like CREATE TABLE.
      *
-     * @param array $fields
+     * @param array|Index $fields
```

Comment:
This is weird and we should not do this. This API accepts an array of field definitions and not an index. I see why you are doing this but this is more of a workaround/hack rather than a good solution. Also having `lengths` option on the index as an array of lengths for all of the index columns isn't better either. What we are missing is an `IndexColumn` class that has options itself dedicated to the index column but I am not sure we should/can add this in 2.x.